### PR TITLE
Rework factory bootstrap to take next available object instance ID

### PR DESCRIPTION
**Pull request by seank-img**
Signed-off-by: Sean Kelly <Sean.Kelly@imgtec.com>

### CODE REVIEW

File: core/src/common/lwm2m_bootstrap_config.c [hunk 1]
```diff
@@ -203,19 +203,41 @@ void BootstrapInformation_DeleteBootstrapInfo(const BootstrapInfo * bootstrapInf
     free((void *)bootstrapInfo);
 }
 
+AwaObjectInstanceID FindNextFreeObjectInstanceID(Lwm2mContextType * context, AwaObjectID ObjectID, AwaObjectInstanceID StartObjectInstanceID)
+{
+    AwaObjectInstanceID result = AWA_INVALID_ID;
+
+    if(StartObjectInstanceID != AWA_INVALID_ID)
```
Comment (chris-dewbery-imgtec) [inline core/src/common/lwm2m_bootstrap_config.c]: if<space>(

File: core/src/common/lwm2m_bootstrap_config.c [hunk 2]
```diff
@@ -203,19 +203,41 @@ void BootstrapInformation_DeleteBootstrapInfo(const BootstrapInfo * bootstrapInf
...
+        result = StartObjectInstanceID;
+
+    while(Lwm2mCore_Exists(context, ObjectID, result, AWA_INVALID_ID))
+    {
+        result++;
+    }
+
+    if(result > AWA_MAX_ID)
```
Comment (chris-dewbery-imgtec) [inline core/src/common/lwm2m_bootstrap_config.c]: if(

**Comment by chris-dewbery-imgtec** [comment]
+ 1 if the coding convention violations are corrected.

File: core/src/common/lwm2m_bootstrap_config.c [hunk 3]
```diff
@@ -203,19 +203,41 @@ void BootstrapInformation_DeleteBootstrapInfo(const BootstrapInfo * bootstrapInf
     free((void *)bootstrapInfo);
 }
 
+AwaObjectInstanceID FindNextFreeObjectInstanceID(Lwm2mContextType * context, AwaObjectID ObjectID, AwaObjectInstanceID StartObjectInstanceID)
+{
+    AwaObjectInstanceID result = AWA_INVALID_ID;
+
+    if (StartObjectInstanceID != AWA_INVALID_ID)
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_bootstrap_config.c]: Missing { }

File: core/src/common/lwm2m_bootstrap_config.c [hunk 4]
```diff
@@ -203,19 +203,41 @@ void BootstrapInformation_DeleteBootstrapInfo(const BootstrapInfo * bootstrapInf
...
+        result = StartObjectInstanceID;
+
+    while(Lwm2mCore_Exists(context, ObjectID, result, AWA_INVALID_ID))
+    {
+        result++;
+    }
+
+    if (result > AWA_MAX_ID)
```
Comment (david-antliff-imgtec) [inline core/src/common/lwm2m_bootstrap_config.c]: Missing { }

**Comment by david-antliff-imgtec** [comment]
Agreed to merge provided missing braces are fixed in the next PR.

**Pull request merged**